components/resource_control/src/resource_group.rs

```diff
@@ -218,26 +219,43 @@ impl ResourceGroupManager {
         }
     }
 
-    pub fn get_resource_limiter(&self, rg: &str) -> Option<Arc<ResourceLimiter>> {
-        if let Some(g) = self.resource_groups.get(rg) {
-            return g.limiter.clone();
+    pub fn get_resource_limiter(
+        &self,
+        rg: &str,
+        request_source: &str,
+    ) -> Option<Arc<ResourceLimiter>> {
+        if let Some(group) = self.resource_groups.get(rg) {
+            if !group.fallback_default {
+                return group.get_resource_limiter(request_source);
+            }
         }
 
-        // fallback to the default resource group if target group doesn't exist.
         self.resource_groups
             .get(DEFAULT_RESOURCE_GROUP_NAME)
-            .and_then(|g| g.limiter.clone())
+            .and_then(|g| g.get_resource_limiter(request_source))
     }
 }
 
 pub(crate) struct ResourceGroup {
     pub group: PbResourceGroup,
     pub limiter: Option<Arc<ResourceLimiter>>,
+    background_source_types: HashSet<String>,
+    // whether to fallback background resource control to `default` group.
+    fallback_default: bool,
 }
 
 impl ResourceGroup {
     fn new(group: PbResourceGroup, limiter: Option<Arc<ResourceLimiter>>) -> Self {
-        Self { group, limiter }
+        let background_source_types =
+            HashSet::from_iter(group.get_background_settings().get_job_types().to_owned());
+        let fallback_default =
```

Comment:
May we fallabck not default and not itself?

Reply:
Didn't got your point. In the current product design, the background control falls back to the default group's control if the group's background control setting is the default config(not explicitly set). So `fallback_default ` is the flag means this group should use the default group's control.

Reply:
I mean we may make a reference to another group, which may be non-default or the same group. All right, we can modify this later if it's needed.